app/data/models.py

```diff
@@ -28,7 +21,25 @@ class Label(Base):
     id = Column(Integer, primary_key=True)
     label_id = Column(String, nullable=False)
     name = Column(String, nullable=False, unique=True)
-    images = relationship("Image", secondary=image_label, back_populates="labels")
+    images = relationship("Image", secondary="label_assignment")
 
     def __repr__(self):
         return f"Label('{self.id}', '{self.label_id}', '{self.name}')"
+
+
+class LabelAssignment(Base):
+    __tablename__ = "label_assignment"
+    id = Column(Integer, primary_key=True)
+    image_id = Column(String, ForeignKey("image.image_id"))
+    label_id = Column(String, ForeignKey("label.label_id"))
+    origin = Column(String(256), nullable=False)
```

Comment:
I am considering implementing this field as class as well

Reply:
but for now a string is probably fine